tests/baselines/reference/truthinessCallExpressionCoercion2.errors.txt

```diff
@@ -0,0 +1,99 @@
+tests/cases/compiler/truthinessCallExpressionCoercion2.ts(3,5): error TS2774: This condition will always return true since the function is always defined. Did you mean to call it instead?
+tests/cases/compiler/truthinessCallExpressionCoercion2.ts(6,10): error TS2774: This condition will always return true since the function is always defined. Did you mean to call it instead?
+tests/cases/compiler/truthinessCallExpressionCoercion2.ts(30,18): error TS2774: This condition will always return true since the function is always defined. Did you mean to call it instead?
+tests/cases/compiler/truthinessCallExpressionCoercion2.ts(41,5): error TS2774: This condition will always return true since the function is always defined. Did you mean to call it instead?
+tests/cases/compiler/truthinessCallExpressionCoercion2.ts(44,10): error TS2774: This condition will always return true since the function is always defined. Did you mean to call it instead?
+tests/cases/compiler/truthinessCallExpressionCoercion2.ts(60,9): error TS2774: This condition will always return true since the function is always defined. Did you mean to call it instead?
+tests/cases/compiler/truthinessCallExpressionCoercion2.ts(63,14): error TS2774: This condition will always return true since the function is always defined. Did you mean to call it instead?
+
+
+==== tests/cases/compiler/truthinessCallExpressionCoercion2.ts (7 errors) ====
+    function test(required1: () => boolean, required2: () => boolean, optional?: () => boolean) {
+        // error
+        required1 && console.log('required');
+        ~~~~~~~~~
+!!! error TS2774: This condition will always return true since the function is always defined. Did you mean to call it instead?
+    
+        // error
+        1 && required1 && console.log('required');
+             ~~~~~~~~~
+!!! error TS2774: This condition will always return true since the function is always defined. Did you mean to call it instead?
+    
+        // ok
+        required1 && required1();
+    
+        // ok
+        required1 && 1 && required1();
+    
+        // ok
+        optional && console.log('optional');
+    
+        // ok
+        1 && optional && console.log('optional');
+    
+        // ok
+        !!required1 && console.log('not required');
+    
+        // ok
+        required1() && console.log('required call');
+    
+        // ok
+        required1 && required2 && required1() && required2();
+    
+        // error
+        required1 && required2 && required1() && console.log('foo');
+                     ~~~~~~~~~
+!!! error TS2774: This condition will always return true since the function is always defined. Did you mean to call it instead?

```

Comment:
@DanielRosenwasser Do you mean this cases?

Reply:
Yup!